scanner/src/routes/loading/+page.svelte

```diff
@@ -0,0 +1,53 @@
+<script lang="ts">
+  import { onMount } from "svelte"
+  import { page } from "$app/stores"
+  import Loading from "$components/Loading/index.svelte"
+  import { dataProduct } from "$lib/api"
+  import { redirect } from "@sveltejs/kit"
+  import { goto } from "$app/navigation"
+
+  onMount(async () => {
+    // Just an example of fetching a data product
+    const { dataspace_domain, product_path, source, product, id } = history.state
+
+    if (!dataspace_domain || !product_path || !product || !id) {
+      goto("/")
+      return
+    }
+
+    const result = await dataProduct.fetch({
+      dataspace_domain,
+      product_path,
+      source,
+      product,
+      id,
+    }).result
+
+    if (result.ok) {
+      goto(
+        `/q/${encodeURIComponent(dataspace_domain)}/${encodeURIComponent(
+          product_path
+        )}/${encodeURIComponent(id)}`,
+        {
+          replaceState: true,
+          state: result.data,
+        }
+      )
+    } else {
+      console.error("Generating tag failed", result)
+    }
```

Comment:
A whole lot about this is very wrong.

Line 11: This is supposed to *START LOADING METADATA* after scanning the QR code, what are the paramaters for the metadata? Definitely not `dataspace_domain`, it is a response from the metadata request.

Line 13: Where was this logic asked for?

Line 18: Why are you fetching dataproducts? Where was this logic documented?

Line 27: Why are you 1) depending on the result of the metadata query, based on mock data, 2) depending on pointless result of a pointless dataproduct fetch query, 3) and THEN redirecting to the URI which looks almost correct to even start this process, 4) but the URL is even wrong?

Line 37: you're not generating a tag.

Reply:
The URL structure is `/q/{iss}/{product}/{id}` not `/q/{dataspace_domain}/{product_path}/{id}`, these are VERY different things